lib/plugins/aws/lib/naming.test.js

```diff
@@ -94,10 +94,16 @@ describe('#naming()', () => {
   });
 
   describe('#getStackName()', () => {
-    it('should use the service name and stage from the service and config', () => {
+    it('should use the service name and stage from the service and config as default', () => {
       serverless.service.service = 'myService';
       expect(sdk.naming.getStackName()).to.equal(`${serverless.service.service}-${options.stage}`);
     });
+
+    it('should use the provider options to supply a custom stackName', () => {
```

Comment:
We should add another test which also checks if the `stackName` in the `provider` property of `serverless.yml` is also considered.